src/wp-includes/option.php

```diff
@@ -819,6 +815,76 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ * Determines the autoload value for a given option.
+ *
+ * @param string|bool|null  $autoload The autoload value for the option. Can be 'yes', 'no', true, false, or null.
+ * @param string $name                The name of the option.
+ * @param mixed  $value               The value of the option.
+ *
+ * @return string The determined autoload value. Possible values: 'yes', 'no', 'default'.
+ */
+function get_autoload_value( $autoload, $name, $value ) {
+	global $wpdb;
+
+	// Check if autoload is explicitly set to 'no' or false.
+	if ( 'no' === $autoload || false === $autoload ) {
+		return 'no';
+	}
+
+	// Check if autoload is explicitly set to 'yes' or true.
+	if ( 'yes' === $autoload || true === $autoload ) {
+		return 'yes';
+	}
+
+	// Check if autoload is explicitly set to 'default' or true.
+	if ( 'default' === $autoload ) {
+		return $autoload;
+	}
```

Comment:
I think the signature and part of the function is confusing. Most of the lines I marked here should be removed, as it would make more sense API-wise for this to be a function that _only_ deals with which autoload value is set in the database.

I think this function should rather be like `get_autoload_value( $option )` and return either the string from the database field, or something false-y if there is no entry for it in the database.

For everything else this function does we can either introduce utilities or handle it inline in the functions where it is relevant.

Reply:
I degree. This function is follows DRY [principle](https://en.wikipedia.org/wiki/Principle#Principle_as_axiom_or_logical_fundament). Introducing lots of repeated logic in add_option and update_option does make any sense to me. Keeping this logic contained a function, makes this code easier to test and in the future, we could easily add a filter to `get_autoload_value`. Specially we could use this filter, with `register_setting` to define the autoload value. 

I agree the naming of this function could be improved. `get_autoload_value` isn't really correct. Any thoughts?

Reply:
@spacedmonkey You could also use multiple functions in a way that still follows the DRY principle. Currently, the function here violates the single responsibility principle:
* It parses a given autoload value to one that is supported by the database.
* It checks for the option size and forces a different autoload value if too large.
* It gets the current autoload value from the database.

Those should be 3 different functions.